packages/ordeq/tests/resources/resources/custom_partition.py

```
@@ -0,0 +1,24 @@
+# Captures that the same IO can have more than one (shared) resource:
+from dataclasses import dataclass
+from pathlib import Path
+
+from ordeq_files import CSV
+
+
+@dataclass(frozen=True)
+class Partition:
+    idx: str
+
+
+partition_nl = Partition("NL")
+partition_be = Partition("BE")
+partition_eu = Partition("Europe")
+folder = Path("folder")
+
+csv_nl = CSV(path=folder / Path(partition_nl.idx)) @ partition_nl
+csv_be = CSV(path=folder / Path(partition_be.idx)) @ partition_be
+csv_nl.with_resource(partition_eu)
+csv_be.with_resource(partition_eu)
```

Should raise an error